packages/node/index.js

```diff
@@ -198,13 +202,15 @@ export class NodeStackable extends BaseStackable {
         }
       }
 
-      if (this.#isFastify) {
-        this.logger.trace({ injectParams }, 'injecting via fastify')
-        res = await this.#app.inject(injectParams, onInject)
-      } else {
-        this.logger.trace({ injectParams }, 'injecting via light-my-request')
-        res = await inject(this.#dispatcher ?? this.#app, injectParams, onInject)
-      }
+      await this.#asyncResource.runInAsyncScope(async () => {
+        if (this.#isFastify) {
+          this.logger.trace({ injectParams }, 'injecting via fastify')
+          res = await this.#app.inject(injectParams, onInject)
```

Comment:
I think you can avoid this function being async and this `await` by returning `res`, and only awaiting the result. That should remove a few promises from a relatively hot path.